src/main/resources/static/surveyor.js

```diff
@@ -39,10 +39,219 @@ $(document).ready(function() {
         $("#Questions").append(newOpenQuestion)    
     });
 
-    //Should send a POST to the endpoint with the servey details as a JSON
-    $(".createSurvey").click(function(){  
-        //  
+    //Should send a POST to the endpoint with the survey details as a JSON
+    $(".createSurvey").click(async function() {
+
+        //JSON objects initialization
+        var jsonQuestions = [];
+
+        //Carry the error for nested function
+        let error = false;
+
+        //Extract surveyName
+        let surveyName = $('#surveyName').val();
+
+        //Check if survey name is present
+        if(!surveyName) {
+            alert("ERROR: Please specify the name of the survey");
+            error = true;
+            return
+        }
+
+        surveyName = JSON.stringify({"name": surveyName});
+        let questions = $('div[id^="Questions"]');
+
+        //Handle empty questions
+        if (questions.is(':empty')) {
+            alert("ERROR: There must be atleast 1 question in the survey!")
+            error = true;
+            return
+        }
+
+        questions.children().each(function () {
+            //Contains jsonData of the question
+            if(error !== true) {
+                error = addJSONQuestion(jsonQuestions, this)
+                console.log(error)
+            }
+        });
+
+        //Don't do the ajax call if there was an error with one of the questions
+        if (error === true) return
```

Comment:
you arent going to get here, if there was an error you are returning after each one

Reply:
ah actually nvm we do get an error if the question is wrong